wsidicom/instance.py

```diff
@@ -592,6 +396,132 @@ def slice_thickness(self) -> Optional[float]:
         """Return slice thickness."""
         return self._slice_thickness
 
+    @property
+    def wsi_type(self) -> str:
+        return self._get_wsi_flavor(self.ImageType)
+
+    @classmethod
+    def is_supported_wsi_dicom(
+        cls,
+        dataset: Dataset,
+        transfer_syntax: UID
+    ) -> Optional[str]:
+        """Check if dataset is dicom wsi type and that required attributes
+        (for the function of the library) is available.
+        Warn if attribute listed as requierd in the library or required in the
+        standard is missing.
+
+        Parameters
+        ----------
+        dataset: Dataset
+            Pydicom dataset to check if is a WSI dataset.
+        transfer_syntax: UID
+            Transfer syntax of dataset.
+
+        Returns
+        ----------
+        Optional[str]
+            WSI image flavor
+        """
+
+        sop_class_uid = getattr(dataset, "SOPClassUID", "")
+        if sop_class_uid != WSI_SOP_CLASS_UID:
+            return None
+
+        SUPPORTED_IMAGE_TYPES = ['VOLUME', 'LABEL', 'OVERVIEW']
+        image_flavor = cls._get_wsi_flavor(dataset.ImageType)
+        image_flavor_supported = image_flavor in SUPPORTED_IMAGE_TYPES
+        if not image_flavor_supported:
+            warnings.warn(f"Non-supported image type {image_flavor}")
+            return None
+
+        for name, attribute in WSI_ATTRIBUTES.items():
+            if (
```

Comment:
There seems to be no warning when attributes are missing?

Reply:
Re-added the warning.